deps/rcheevos/src/rc_compat.c

```diff
@@ -3,6 +3,10 @@
 #include <ctype.h>
 #include <stdarg.h>
 
+#if defined(_3DS) && !defined(_POSIX_THREADS)
+#include <../rthreads/ctr_pthread.h>
+#endif
+
```

Comment:
`deps/rcheevos/src/rc_compat.c` refers to functions from the `pthread_mutex_` family and needs their prototypes. In a griffin build those become available due to header pollution from `libretro-common/rthreads/rthreads.c` (declaring them static is meaningless in this mode).

I got my devkitpro fairly recently and the prototypes are there (guarded behind ifdef `_POSIX_THREADS`), so I didn't see the problem with rc_compat.c until I saw the build failure on CI. I assume it's got an older version of the toolchain, where the functions are not implemented yet, or at least their prototypes are not exposed.

I don't see a better solution to this problem that wouldn't require a slightly invasive rewrite. Except of course updating the toolchain used in the CI infrastructure.

Reply:
Sorry for the noise, I saw your name in git blame output without realizing what b7ddac2431f was about.